Conch/lib/Conch/Plugin/Rollbar.pm

```diff
@@ -0,0 +1,112 @@
+=pod
+
+=head1 NAME
+
+Conch::Plugin::Rollbar
+
+=head1 DESCRIPTION
+
+Mojo plugin to send exceptions to L<Rollbar|https://rollbar.com>
+
+=head1 METHODS
+
+=cut
+
+package Conch::Plugin::Rollbar;
+
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+use Sys::Hostname;
+
+use constant ROLLBAR_ENDPOINT => 'https://api.rollbar.com/api/1/item/';
+
+=head2 register
+
+Adds `send_exception_to_rollbar` to Mojolicious app
+
+=cut
+
+sub register ( $self, $app, $conf ) {
+	$app->helper( send_exception_to_rollbar => sub { _record_exception(@_) } );
+}
+
+# asynchronously send exception details to Rollbar if 'rollbar_access_token' is
+# specified in the commit
+sub _record_exception ( $c, $exception ) {
+	my $access_token = $c->app->config('rollbar_access_token') || return;
+	my $environment  = $c->app->config('rollbar_environment')  || $c->app->mode;
+
+	my @frames = map {
+		{
+			class_name => $_->[0],
+			filename   => $_->[1],
+			lineno     => $_->[2],
+			method     => $_->[3],
+		}
+	} $exception->frames->@*;
+
+	my $context = {
+		context => {
+			pre  => $exception->lines_before,
+			post => $exception->lines_after
+		}
+	};
+
+	# We only have context for the first frame
+	$frames[0]->{context} = $context;
+
+	my $version = `git rev-parse HEAD`;
```

Comment:
I'd like to avoid executing git for every exception we encounter. I'd rather put this into a variable on startup.

Reply:
I don't think performance is too much of a concern.
```
❯ perl -e'
use Benchmark q/timethis/;

timethis(1_000, q/`git rev-parse HEAD`/);
'
timethis 1000:  3 wallclock secs ( 0.06 usr  0.27 sys +  1.27 cusr  0.58 csys =  2.18 CPU) @ 458.72/s (n=1000)
```

We shouldn't be throwing so many exceptions that this code path is executed frequently in any case. I don't think how responsively a user gets their impending 500 'Something went wrong' is a big issue.

If still concerned, let's gather all instances of git executions and put them in the same plugin ( `Conch::Plugin::Git` or w/e) in a separate PR so the indirections are all in the same place.

Reply:
@sungo Went ahead and made the change by pulling it up to the register method.

I threw together a GitVersion plugin to pull all the git-version-y things in one place. If you thing it's worthwhile, I'll throw it in a PR.
https://github.com/joyent/conch/compare/user/lane/git-version-plugin

Reply:
Ok, wasn't sure. Will put the plugin in a separate PR (and update the other whichever gets merged first).